source/val/validate_image.cpp

```diff
@@ -887,6 +887,7 @@ spv_result_t ValidateTypeSampledImage(ValidationState_t& _,
   // Vulkan uses the Sampled=1 case.
   if ((info.sampled != 0) && (info.sampled != 1)) {
     return _.diag(SPV_ERROR_INVALID_DATA, inst)
+           << _.VkErrorID(4659)
```

Comment:
This doesn't seem right. This code validates OpTypeSampledImage, not the opcodes listed in the VU. This VU is already checked in `ValidateImageQuerySizeLod` and `ValidateImageQueryLevels`.

Reply:
the issue is without this the `OpImageQueryLod` usage fails (`ValidateImage.QueryLodVulkanNonSampledError`)

As you pointed [out here](https://github.com/KhronosGroup/SPIRV-Tools/pull/4103#discussion_r556545291), should there now be an explicit check in `ValidateImageQueryLod()`?

Reply:
What about changing this check slightly to cover part of VUID 04657 instead? My concern with using 04659 is that it mentions specific opcodes which may not be involved at all.

Reply:
agree